Tests/JobsTests/JobSchedulerTests.swift

```diff
@@ -152,7 +152,7 @@ final class JobSchedulerTests: XCTestCase {
         jobSchedule.addJob(TriggerShutdownParameters(), schedule: .everyMinute(second: dateComponents.second!))
 
         await withThrowingTaskGroup(of: Void.self) { group in
-            let serviceGroup = ServiceGroup(
+            let serviceGroup = await ServiceGroup(
```

Comment:
Is it possible that `setMetada` or `getMetada` could be called before a persist driver is ready?

Reply:
We aren't using the persist driver as it would pull in the Hummingbird repository as a dependency. I guess you mean before the Postgres client has initialised? I have a fix for this, in hb-postgres. It is a bit of a hack, but basically I wait 100 microseconds to ensure the Postgres client has started up.

Reply:
Yes, I meant before the Postgres client has initialized. `It is a bit of a hack, but basically I wait 100 microseconds to ensure the Postgres client has started up` This could be flaky, should this where application might need restart incase the Postgres client takes more 100 microseconds to initialize?

Reply:
The thing is queries don't fail. It's just the connections manager isn't running. It's a pain because the error is basically false in this situation because the connnection manager has been started it just hasn't got to the point where it has set the flag saying it is running.

Reply:
> The thing is queries don't fail. It's just the connections manager isn't running. It's a pain because the error is basically false in this situation because the connnection manager has been started it just hasn't got to the point where it has set the flag saying it is running.

I see! So, migrations would still apply without any issues and mediate would get saved?

Reply:
> see! So, migrations would still apply without any issues and mediate would get saved?

Yes

Reply:
Thank You! I don't have anymore questions on this PR. Also, please mark this conversion as resolved. I don't any options mark as resolved on my end.